### PR TITLE
PCFG-321 creating massive ICAs upload

### PR DESCRIPTION
**Pull request by FedericoRuzzier**
#### List of Changes

Adding massive ICAs upload from zip file

#### How Has This Been Tested?

Testing was done using unit tests included in pr

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/pagopa/pagopa/apiconfig/service/IcaService.java [hunk 1]
```diff
@@ -150,23 +152,63 @@ public XSDValidation verifyXSD(MultipartFile xml) {
...
+    private void createIca(@NotNull InputStream inputStream, Boolean force, boolean massive) throws SAXException, IOException{
+        List<CheckItem> checks = verifyIca(inputStream, force);
 
         Optional<CheckItem> check = checks.stream()
                 .filter(item -> item.getValid()
                         .equals(CheckItem.Validity.NOT_VALID))
                 .findFirst();
         if (check.isPresent()) {
-            throw new AppException(AppError.ICA_BAD_REQUEST, String.format("[%s] %s", check.get().getValue(), check.get().getNote()));
+            if(!massive) {
```
Comment (aomegax) [inline src/main/java/it/pagopa/pagopa/apiconfig/service/IcaService.java:196]: No, if there is at least an error massive upload will be stopped, so there is no reason to add massive boolean.
Besides I think it is better to deny force upload for massive functionality.

File: src/main/java/it/pagopa/pagopa/apiconfig/service/IcaService.java [hunk 2]
```diff
@@ -150,23 +152,63 @@ public XSDValidation verifyXSD(MultipartFile xml) {
...
+        }catch(SAXException | IOException e){
+            throw new AppException(HttpStatus.BAD_REQUEST, ICA_BAD_REQUEST, "Problem when creating new ICA");
+        }
+    }
+
+    @Transactional
+    @java.lang.SuppressWarnings({"javasecurity:S6096", "java:S5443"})
+    public void createMassiveIcas(@NotNull MultipartFile file, Boolean force) {
+        try{
+            ZipInputStream zis = new ZipInputStream(file.getInputStream());
```
Comment (aomegax) [inline src/main/java/it/pagopa/pagopa/apiconfig/service/IcaService.java:166]: could these code lines be generalized considering massive verification icas too?

**Review by aomegax — CHANGES_REQUESTED** [review]

File: src/main/java/it/pagopa/pagopa/apiconfig/controller/IcaController.java [hunk 3]
```diff
@@ -177,4 +177,23 @@ public ResponseEntity<List<MassiveCheck>> massiveVerifyIcas(
...
+            @ApiResponse(responseCode = "404", description = "Not Found", content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE, schema = @Schema(implementation = ProblemJson.class))),
+            @ApiResponse(responseCode = "429", description = "Too many requests", content = @Content(schema = @Schema())),
+            @ApiResponse(responseCode = "500", description = "Service unavailable", content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE, schema = @Schema(implementation = ProblemJson.class)))})
+    @PostMapping(
+            value = "/massive",
+            consumes = {MediaType.MULTIPART_FORM_DATA_VALUE}
+    )
+    public ResponseEntity<Void> massiveCreateIcas(@NotNull @Parameter(description = "Zip file containing ICAs to create", required = true, content = @Content(mediaType = MediaType.APPLICATION_OCTET_STREAM_VALUE)) @RequestParam("file") MultipartFile file,
+                                          @Parameter(description = "Force upload ignoring the validity date") @RequestParam(value = "force", defaultValue = "false") Boolean force) {
```
Comment (aomegax) [inline src/main/java/it/pagopa/pagopa/apiconfig/controller/IcaController.java:195]: no `force` parameter

**Review by aomegax — CHANGES_REQUESTED** [review]

File: src/main/java/it/pagopa/pagopa/apiconfig/service/IcaService.java (hunk 2 above)
Comment (FedericoRuzzier) [reply src/main/java/it/pagopa/pagopa/apiconfig/service/IcaService.java:166]: I'm trying, problem is two different functions (create and verify) are called in the two methods inside the same condition

**Review by jacopocarlini — APPROVED** [review]
🏅 Good Job!

**Review by aomegax — APPROVED** [review]
👍 good job!

**Pull request merged**